### PR TITLE
dan/per-11149-add-ruff-to-pdp-repo

### PR DESCRIPTION
**Pull request by danyi1212**

### CODE REVIEW

File: README.md
Comment (omer9564) [inline README.md:28]: PDP API KEY is required here

File: README.md
Comment (omer9564) [inline README.md:34]: I don't think there is really an AUTHZ_SERVICE_URL - PDP_CONTROL_PLANE is the right one

File: scripts/gunicorn_conf.py
Comment (danyi1212) [reply scripts/gunicorn_conf.py:4]: I think it's good without, but will add anyway

File: horizon/enforcer/utils/mapping_rules_utils.py
Comment (danyi1212) [reply horizon/enforcer/utils/mapping_rules_utils.py:54]: https://docs.astral.sh/ruff/rules/in-dict-keys/

File: horizon/enforcer/api.py
Comment (danyi1212) [reply horizon/enforcer/api.py]: Nothing used the `processed_query`, and `get_v1_processed_query` has no side-effects. It can be safely removed

File: scripts/gunicorn_conf.py
Comment (danyi1212) [reply scripts/gunicorn_conf.py:4]: Never mind, ruff format removes it. It's not needed

File: horizon/enforcer/api.py
Comment (omer9564) [inline horizon/enforcer/api.py:482]: why remove this ?

Comment (omer9564) [reply horizon/enforcer/api.py:482]: don't think we use this but anyway

File: requirements.txt
Comment (omer9564) [inline requirements.txt]: dangling dependency I guess ?

File: requirements-dev.txt
Comment (omer9564) [inline requirements-dev.txt]: worth adding a comment in the readme then that for development, both are required to be installed

File: horizon/enforcer/api.py
Comment (omer9564) [reply horizon/enforcer/api.py:445]: this is still relevant

File: horizon/enforcer/api.py
Comment (omer9564) [reply horizon/enforcer/api.py:738]: still waiting for answer here

**Review by omer9564 — COMMENTED** [review]
few more comments, and there are some leftovers from previous review.
Anyway overall looks good

File: requirements.txt
Comment (danyi1212) [reply requirements.txt]: I've added it and went back from using it

File: horizon/enforcer/api.py
Comment (danyi1212) [reply horizon/enforcer/api.py:482]: It's only removing the x_permit_sdk_language as it was not used anywhere

File: horizon/enforcer/api.py
Comment (danyi1212) [reply horizon/enforcer/api.py:738]: The `raw_result` didn't exist. I've looked into the git history and found it in this commit
https://github.com/permitio/PDP/commit/8f5325c9491601f1cb01f96bd1256a19ac1a7ba0

Comment (omer9564) [reply horizon/enforcer/api.py:738]: Oh I see now, you basically say that this API always got to the except block due to a reference to non existing variable for almost 4 years, right

File: horizon/enforcer/api.py
Comment (omer9564) [reply horizon/enforcer/api.py]: Only left thing is that, I'll approve so you can fix it without needing another review, but pretty sure there is an edge case here ( that is probably never reached but still exists )

**Review by omer9564 — APPROVED** [review]

**Pull request merged**
